sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets

```diff
@@ -124,14 +124,37 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
   </Target>
 
   <!-- This target is used to set up the run arguments for the function app, supporting `dotnet run` -->
-  <Target Name="_FunctionsComputeRunArguments" BeforeTargets="ComputeRunArguments">
+  <Target Name="_FunctionsComputeRunArguments" BeforeTargets="ComputeRunArguments" DependsOnTargets="_CheckForFuncInstallation" Condition="'$(FuncExists)' == 'true'">
     <PropertyGroup>
       <RunCommand>func</RunCommand>
       <RunArguments>host start $(RunArguments)</RunArguments>
       <RunWorkingDirectory>$(OutDir)</RunWorkingDirectory>
     </PropertyGroup>
   </Target>
+
+  <Target Name="_CheckForFuncInstallation">
+    <PropertyGroup>
+      <PathCheckCommand Condition="'$(OS)' == 'Windows_NT'">where func</PathCheckCommand>
+      <PathCheckCommand Condition="'$(OS)' != 'Windows_NT'">which func</PathCheckCommand>
+    </PropertyGroup>
+    <Exec Command="$(PathCheckCommand)" IgnoreExitCode="true">
+      <Output TaskParameter="ExitCode" PropertyName="FuncExitCode" />
+    </Exec>
+    <PropertyGroup>
+      <FuncExists Condition="'$(FuncExitCode)' == '0'">true</FuncExists>
+    </PropertyGroup>
+  </Target>
   
+  <Target Name="_FunctionsComputeRunArgumentsWhenFuncNotExists" BeforeTargets="ComputeRunArguments" DependsOnTargets="_CheckForFuncInstallation" Condition="'$(FuncExists)' != 'true'">
+    <PropertyGroup>
+      <ErrorMessage>Azure Functions Core Tools is missing. Please install it and try again. Please visit https://aka.ms/azfunc-coretools-not-installed for more details.</ErrorMessage>
+      <RunCommand Condition="'$(OS)' == 'Windows_NT'">cmd.exe</RunCommand>
```

Comment:
Can we hook into msbuild targets and just use `Error` task here instead of passing to `RunCommand`?

Also, I wonder if we can use `cmd` and `sh` to invoke `func` installed via npm or the like?

Reply:
We want to show this error when user executes the `dotnet run` command. So we need to rely on the "ComputeRunArguments" target before hook. The runtime code tries to execute whatever is passed via `RunCommand` and `RunArguments`. Using `<Error>` also works but produces a failure with the targets file info etc. See below.

![image](https://github.com/user-attachments/assets/cb814acc-3868-4b05-a967-3ae7d02b6f13)

With the RunCommand approach, it's much nicer.

![image](https://github.com/user-attachments/assets/7d451d8a-f985-4139-b421-ecebd4b07e37)

I felt the second one is better. But open to suggestions. @mattchenderson Thoughts?

Let me check the npm install use case.

Reply:
I like the first one personally - it catches the attention and makes it clear we never got to the point of even running the application.

Reply:
It appears that `dotnet run` does not work today when core tools is installed via npm.

![image](https://github.com/user-attachments/assets/4a5749eb-9779-4b20-95f5-a6e553f8ad63)

Reply:
@kshyju yes there was a customer reported issue where we found that problem. The problem is that npm doesn't install a func executable - it installs func scripts, like func.cmd, func.ps1, func.sh. These are not supported by dotnet run by default. I think we can work around this by using `cmd` or `bash` (OS dependent) as the process, with `func` as an argument.

I did a similar thing for our azurite fixture: https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/Fixtures/AzuriteFixture.cs#L152